internal/objects/crdrules.go

```diff
@@ -178,12 +237,35 @@ func (c *CRDLister) UpdateFqdnHTTPRulesMappings(fqdn, path, httprule string) {
 }
 
 // FqdnSharedVSModelCache/SharedVSModelFqdnCache
-func (c *CRDLister) GetFQDNToSharedVSModelMapping(fqdn string) (bool, string) {
-	found, modelName := c.FqdnSharedVSModelCache.Get(fqdn)
-	if !found {
-		return false, ""
+func (c *CRDLister) GetFQDNToSharedVSModelMapping(fqdn string) (bool, []string) {
+	oktype, fqdnType := c.FqdnFqdnTypeCache.Get(fqdn)
+	if !oktype || fqdnType == "" {
+		fqdnType = string(akov1alpha1.Exact)
+	}
+
+	allFqdns := c.FqdnSharedVSModelCache.GetAllKeys()
+	returnModelNames := []string{}
+	for _, mFqdn := range allFqdns {
+		if fqdnType == string(akov1alpha1.Exact) && mFqdn == fqdn {
+			if found, modelName := c.FqdnSharedVSModelCache.Get(mFqdn); found {
```

Comment:
if the fqdn is not found in the Cache, we would break here and then return true in line no 268 with empty list. Is that fine ?